src/v2/segment.v2.ejs

```diff
@@ -0,0 +1,159 @@
+<%% if (!disable.segment && !sd.THIRD_PARTIES_DISABLED && sd.SEGMENT_WRITE_KEY) { %>
+<script type="text/javascript" defer>
+
+!function(){var analytics=window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","debug","page","once","off","on","addSourceMiddleware","addIntegrationMiddleware","setAnonymousId","addDestinationMiddleware"];analytics.factory=function(t){return function(){var e=Array.prototype.slice.call(arguments);e.unshift(t);analytics.push(e);return analytics}};for(var t=0;t<analytics.methods.length;t++){var e=analytics.methods[t];analytics[e]=analytics.factory(e)}analytics.load=function(t,e){var n=document.createElement("script");n.type="text/javascript";n.async=!0;n.src="https://cdn.segment.com/analytics.js/v1/"+t+"/analytics.min.js";var a=document.getElementsByTagName("script")[0];a.parentNode.insertBefore(n,a);analytics._loadOptions=e};analytics.SNIPPET_VERSION="4.1.0";
+
+// adaptation of:
+//   https://gist.github.com/sperand-io/4725e248a35d5005d68d810d8a8f7b29
+//   https://github.com/littledata/shopify-tracker/blob/ec4fac54cd56344980aa481800ee8dba550ea97d/src/customerPrivacyAPI/oneTrust.ts#L31
+
+const SEGMENT_WRITE_KEY = "<%%= sd.SEGMENT_WRITE_KEY %>";
+const SEGMENT_MAPPING = 'C0001'; // Segment itself assigned OneTrust Strictly Necessary category.
+let SEGMENT_DESTINATIONS = [];
+
+fetchDestinations(SEGMENT_WRITE_KEY).then(
+  destinations => {
+    SEGMENT_DESTINATIONS = destinations;
+
+    waitForOneTrust();
+
+    if (oneTrustReady()) {
+      getConsentAndLoadSegment();
+    }
+  }
+)
+
+// OneTrust calls OptanonWrapper when there's a consent change.
+window.OptanonWrapper = getConsentAndLoadSegment;
+
+function waitForOneTrust() {
+  let attempts = 0;
+  const maxAttempts = 100;
+
+  // wait 1sec at most.
+  if (!oneTrustReady()) {
+    const interval = setInterval(() => {
+      if (oneTrustReady() || attempts > maxAttempts) clearInterval(interval);
+      attempts++;
+    }, 10);
+  }
+}
+
+function oneTrustReady() {
+    // OneTrust is ready if OnetrustActiveGroups contains at least Strictly Necessary group.
+    if ((typeof window.OnetrustActiveGroups === 'string') && (window.OnetrustActiveGroups.split(',').includes('C0001'))) {
```

Comment:
Observation: On page load, `OnetrustActiveGroups` is `,,` (just commas). It looks like after OneTrust banner script loaded, the string becomes `,C0001,`.